lib/OpenQA/Utils.pm

```diff
@@ -164,6 +166,35 @@ sub testcasedir {
     return $dir;
 }
 
+sub gitrepodir {
+    my %args = (
+        distri  => '',
+        version => '',
+        repo    => 'tests',
+        @_,
+    );
+    my $path
+      = $args{repo} eq 'needles'
+      ? needledir($args{distri}, $args{version})
+      : testcasedir($args{distri}, $args{version});
+    my $filename = '';
+    if (-e path($path, '.git')) {
+        $filename = path($path, '.git', 'config');
+    }
+    else { log_warning("$path is not a git directory"); }
+
+    my $giturl = '';
+    my $config = Config::Tiny->read($filename, 'utf8');
+    unless (defined $config) {
+        return $config->{'remote "origin"'}{url} if ($config->{'remote "origin"'}{url} =~ /^http(s?)/);
+        my @url_tokenized = split(':', $config->{'remote "origin"'}{url});
+        $url_tokenized[1] = $url_tokenized[1] =~ s/\.git/\/commit\//r;
+        my @githost = split('@', $url_tokenized[0]);
+        $giturl = 'https://' . $githost[1] . '/' . $url_tokenized[1];
+    }
+    return $giturl;
```

Comment:
```suggestion
    my $config = Config::Tiny->read($filename, 'utf8');
    return '' if defined $config;
    return $config->{'remote "origin"'}{url} if ($config->{'remote "origin"'}{url} =~ /^http(s?)/);
    my @url_tokenized = split(':', $config->{'remote "origin"'}{url});
    $url_tokenized[1] = $url_tokenized[1] =~ s/\.git/\/commit\//r;
    my @githost = split('@', $url_tokenized[0]);
    return 'https://' . $githost[1] . '/' . $url_tokenized[1];
```

but why would you want to return '' if the config is actually defined? Shouldn't it be the other way around?

Reply:
That one looks a bit weird:
```
$url_tokenized[1] = $url_tokenized[1] =~ s/\.git/\/commit\//r;
```
Why not this:
```
$url_tokenized[1] =~ s/\.git/\/commit\//;
```
or to improve readability:
```
$url_tokenized[1] =~ s{\.git}{/commit/};
```
And I probably would anchor the regex to the string end:
```
$url_tokenized[1] =~ s{\.git$}{/commit/};
```

Reply:
*this change is ready for review*
thanks @perlpunk @okurz